mlibrary/search
# [LIBSEARCH-938] Investigate whether CSS can be used to hide "search only" holdings across all bib records

# Overview
Without touching Spectrum, we need to figure out a way how to hide all Catalog holdings that have `Search only (no full text)` while `Remove search-only HathiTrust materials` is checked. This pull request filters out the incoming holdings when true.

This pull request resolves [LIBSEARCH-938](https://mlit.atlassian.net/browse/LIBSEARCH-938).

## Anything else?
The `Holders` component has been simplified, removing the need for `preExpandedIds` and `createId`.

### `ResourceAccessLoading`
`ResourceAccessLoading` was a very simple component, only ever called once in `ResourceAccessContainer`. The code was copied directly into the container, and the original file was deleted.

### `ContextProvider`
`ContextProvider` was a class component, and has been converted to a functional component to align with [current React practices](https://react.dev/learn/your-first-component).

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Make a [Catalog search](http://localhost:3000/catalog?query=birds&library=All+libraries&filter.search_only=false) with `Remove search-only HathiTrust materials` unselected.
  - Do you see any `HathiTrust Digital Library` holdings that show `Search only (no full text)`?
  - If you select `Remove search-only HathiTrust materials`, do you not see any `HathiTrust Digital Library` holdings that show `Search only (no full text)`?
